modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/model/ComponentIdentifier.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.config.spring.dsl.model;
+
+import static org.mule.runtime.config.spring.dsl.processor.xml.CoreXmlNamespaceInfoProvider.CORE_NAMESPACE_NAME;
+import org.mule.runtime.core.util.Preconditions;
+
+/**
+ * Unique identifier for a configuration option.
+ *
+ * @since 4.0
+ */
+public class ComponentIdentifier
+{
+
+    private String namespace;
```

Comment:
can't be final since are used by the builder. Anyway there's no way to change them.